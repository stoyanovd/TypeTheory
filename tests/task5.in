(((\i0.\z.\t.\f.(\z0.\f0.\s0.\p0.\p1.(\p.(\m.(\m1.((\f.\x.(f (x x)) (f (x x))) (\g.\a.\b.(z0 a) (t) ((z0 b) f (g (m1 a) (m1 b)))))) (\n.f0 (n (\p.p0 (s0 p) ((p1) (s0 p))) (p0 z z)))) (\m.\n.m (p n) z)) (\m.\n.(m (p1)) n)) (\n.n (\x.f) t) (\a.a t) (\a.a f) (\a.\b.\s.s a b) (\n.\f.\x.n f (f x)))) (\c.\t.\e.c t e) (\f.\x.x) (\a.\b.a) (\a.\b.b)) (\f.\x.f (f x)) (\f.\x.f (f x))