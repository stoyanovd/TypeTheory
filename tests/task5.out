(((S (K (S ((S I) (K ((S (K (S (K (S (K ((S I) (K (K I))))))))) ((S ((S (K S)) ((S (K (S (K S)))) ((S (K (S (K K)))) ((S ((S (K S)) ((S (K K)) I))) (K ((S ((S (K S)) ((S (K (S (K ((S ((S (K S)) ((S (K K)) ((S (K S)) ((S (K (S I))) ((S (K K)) I)))))) (K ((S (K K)) I))))))) ((S ((S (K S)) ((S (K K)) I))) (K ((S (K ((S I) (K ((S (K K)) I))))) I)))))) (K ((S (K ((S I) (K ((S (K K)) I))))) I))))))))) (K (K ((S ((S (K ((S ((S (K S)) ((S (K K)) ((S (K S)) ((S (K (S I))) ((S (K K)) I)))))) (K ((S (K K)) I))))) I)) I)))))))))) ((S (K K)) I)) ((S ((S (K S)) ((S (K K)) I))) ((S ((S (K S)) ((S (K K)) I))) ((S ((S (K S)) ((S (K K)) I))) ((S ((S (K S)) ((S (K K)) I))) ((S ((S (K S)) ((S (K K)) I))) (K I))))))) ((S ((S (K S)) ((S (K K)) I))) ((S ((S (K S)) ((S (K K)) I))) ((S ((S (K S)) ((S (K K)) I))) (K I))))
